NSS-Day-Cohort-53/tabloid-full-stack-amicably-profound-insects
# src-userProfiles list view and details view

#### Changes Made
1. Added file `quill-logo.png` to `client/src/images/` directory.
2. Added file `UserProfileManager.js` to `client/src/modules/` directory. Includes fetch calls - `getAllUserProfiles` and `getUserProfile`.
3. Added file `UserProfilesList.js` to `client/src/components/` directory. Includes view of a list of all active user profiles.
4. Added file `UserProfileListItem.js` to `client/src/components/` directory. Includes a single list item of a profile to be used in `UserProfilesList.js`
5. Added file `UserProfileDetails.js` to `client/src/components/` directory. Includes all relevant details of a single user.

1. Modified file `UserProfileRepository.cs and IUserProfileRepository.cs` to include `GetById and GetAll` methods and functionality.
2. Modified file `UserProfileController.cs` to include `Get() and Get(int id)` methods and functionality.
6. Modified file `UserProfile.cs` to include `CreateDateTimeFormatted` computed property.
7. Modified file `ApplicationViews.js` to include `Routing to User Profiles List and Details and 404 `.
​
#### Steps to Review
1. Checkout this branch locally.
    ```
    git fetch --all
    git checkout src-userProfiles
    ```
8. Open a new Terminal tab (⌘T) and navigate to the server directory.
9. Test app functionality.
    > Log in as a user
    > click `User Profiles` in the header
    > verify that all of the users show up in order of display name (alphabetically)
    > click details of a few users
    > verify that all of the info as required in issue ticket #33 shows
    > when you go to a url of something like `/userprofiles/details/10000` aka the detail page of a user that does not exist, you should see a "404 not found".
10. View code file.
    > Confirm file modifications are present as indicated above.
    > Confirm no unused code or extraneous comments exist.

closes #34 
closes #33
